src/diffr_lib/mod.rs

```diff
@@ -663,25 +663,29 @@ pub fn tokenize(src: &[u8], ofs: usize, tokens: &mut Vec<Span>) {
             tokens.push((lo, hi))
         }
     };
+    let mut kind = GraphemeKind::Other;
     let mut lo = ofs;
-    let mut kind = TokenKind::Other;
     #[allow(clippy::needless_range_loop)]
-    for hi in ofs..src.len() {
+    for (grapheme_start, _, g) in src[ofs..].grapheme_indices() {
+        let hi = grapheme_start + ofs;
         let oldkind = kind;
-        kind = classify_byte(src[hi]);
-        if kind != oldkind || oldkind == TokenKind::Other {
+        kind = classify_grapheme(g);
+        if kind != oldkind || oldkind == GraphemeKind::Other {
             push(lo, hi);
             lo = hi
         }
     }
     push(lo, src.len());
 }
 
-fn classify_byte(b: u8) -> TokenKind {
-    match b {
-        b'a'..=b'z' | b'A'..=b'Z' | b'0'..=b'9' | b'_' => TokenKind::Word,
-        b'\t' | b' ' => TokenKind::Spaces,
-        _ => TokenKind::Other,
+fn classify_grapheme(g: &str) -> GraphemeKind {
+    let first_char = g.chars().next().unwrap_or_default();
+    if first_char.is_alphanumeric() {
+        GraphemeKind::Word
+    } else if first_char == ' ' || first_char == '\t' {
```

Comment:
nitpick: if we change the other branch, we might as well do: 

```suggestion
    } else if first_char.is_whitespace() {
```

Reply:
That's how I had it originally, but then it broke the tests :joy: Problem with `is_whitespace` here is that main.rs uses a `== b' ' || == b'\t'` check to skip leading whitespace, and I'm not too confident about the inter-working of these two.

Reply:
I like your objections against the suggestion, let's leave it as-is